pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -533,14 +533,16 @@ def boxplot_frame_groupby(
         )
         axes = flatten_axes(axes)
 
-        ret = pd.Series(dtype=object)
-
+        data = {}
         for (key, group), ax in zip(grouped, axes):
             d = group.boxplot(
                 ax=ax, column=column, fontsize=fontsize, rot=rot, grid=grid, **kwds
             )
             ax.set_title(pprint_thing(key))
-            ret.loc[key] = d
+            # GH 14701 refactored to allow the 'key' to be passed as a tuple,
+            # which occurs when there is more than one group
+            data[key] = d
+            ret = pd.Series(data)
```

Comment:
We don't need to be creating `ret` on every loop, only after the for loop is done. This line can be dedented.